reprozip-gui/gui.py

```diff
@@ -0,0 +1,461 @@
+        self.master.runName.focus_set()
+        self.master.runName.grid(row = 0, column=1,  sticky = "nsew" )
+        # Asking for the Name of Run
+        labelRunName = tk.Label(self, text = "Name of Run",  font = "Helvetica 14", width = "15")
+        labelRunName.grid(row = 0, column = 0, sticky = "nsew", padx = (30,0))
+        # Textbox for Terminal using Text
+        textTerminal = tk.Text(self, width = "40",height = "8", fg="black" )
+        textTerminal.grid(row = 2, columnspan = 2, pady = (20,0) ,sticky = "nsew", padx = (30,0))
+        # Browse button
+        browseButton = tk.Button(self, text="Browse",command = self._askopenfile, highlightbackground="#404040", highlightthickness="0",fg='white', height ="2", width = "8")
+        browseButton.grid(row = 4, pady = (20,0), sticky = "nw", padx = (30,0))
+        # Display file path
+        workDir = tk.Label(self, textvariable =self.tempPath, fg="white", font = "Helvetica 11 italic", width ="40")
+        workDir.grid(row = 3, columnspan = 2, pady = (20,0),sticky = "nsew", padx = (30,0))
+        # Trace button
+        traceButton = tk.Button(self, text="Trace", command= self._reproTrace,highlightbackground="#404040", highlightthickness="0",fg='white', height ="2", width = "8")
+        traceButton.grid(row = 4, column =1, pady = (20,0), sticky = "ne")
+        
+    def _askopenfile(self):
+        path = tk.filedialog.askopenfilename()
+        self.tempPath.set(path)
+        #print(self.master.runName.get())
+        
+    def _reproTrace(self):
+        tempCommand = "ls"
+        p_status = subprocess.Popen(tempCommand, stdout=subprocess.PIPE, shell=True)
+        (output, err) = p_status.communicate()  
+        #This makes the wait possible
+        p_status.wait()
+        global flag
+        flag = 0
+        self.master.switch_frame(AddRunWindow)
+
+class AddRunWindow(tk.Frame):
+    
+    def __init__(self, master):
+        tk.Frame.__init__(self, master)
+        # Button for adding a new run 
+        addRunButton = tk.Button(self, text = "Add Run", command= self._addRun, highlightbackground="black", highlightthickness="0",fg='white', height ="2", width= "10")
+        addRunButton.grid(row = 1, column =2, padx= (40,0))
+        # Button to edit configuration window 
+        nextButton = tk.Button(self, text=" Next ", command = lambda : master.switch_frame(editConfigurationWindow), highlightbackground="black", highlightthickness="0",fg='white', height ="2", width= "10")
+        nextButton.grid(row = 3, column =2 ,padx= (40,0))
+        # Label for naming runs
+        runLabel = tk. Label(self, text = "Existing Runs",  font = "Helvetica 14")
+        runLabel.grid(row = 0, column =0, sticky= "nw", pady = (10,5))
+        # to read the runs from yaml file
+        with open(self.master.filepath) as fr:
+            data = yaml.safe_load(fr)  
+           
+        if(flag != 3):
```

Comment:
This is really a hack. You shouldn't be passing this through the attribute on the application, rather it should be set by `TraceWindow` before switching.